src/views/PrivateRoutes.jsx

```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from "react";
+import { Navigate, Outlet } from "react-router-dom";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "../../firebase_config";
+
+function PrivateRoutes() {
+  const [user, setUser] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
```

Comment:
maybe simplify this callback by directly setting the user: 
`onAuthStateChanged(auth, (currentUser) => {
  setIsLoading(false);
  setUser(currentUser);
});
`
NOTE: if `currentUser` is `null`, it means the user is not authenticated. If it's an object, it represents the authenticated user.